archiver.go

```diff
@@ -131,131 +131,161 @@ func GetCurrentArchives(ctx context.Context, db *sqlx.DB, org Org, archiveType A
 	return existingArchives, nil
 }
 
+// between is inclusive on both sides
+const lookupOrgDailyArchivesForDateRange = `
+SELECT id, start_date, period, archive_type, hash, size, record_count, url, rollup_id
+FROM archives_archive
+WHERE org_id = $1 AND archive_type = $2 AND period = $3 AND start_date BETWEEN $4 AND $5
+ORDER BY start_date asc
+`
+
+// GetCurrentArchives returns all the current archives for the passed in org and record type
+func GetOrgDailyArchivesForDateRange(ctx context.Context, db *sqlx.DB, org Org, archiveType ArchiveType, startDate time.Time, endDate time.Time) ([]*Archive, error) {
+	existingArchives := []*Archive{}
+	err := db.SelectContext(ctx, &existingArchives, lookupOrgDailyArchivesForDateRange, org.ID, archiveType, DayPeriod, startDate, endDate)
+	if err != nil && err != sql.ErrNoRows {
+		return nil, err
+	}
+
+	return existingArchives, nil
+}
+
+const lookupMissingDailyArchive = `
+WITH month_days(missing_day) AS (
+  select generate_series($1::timestamp with time zone, $2::timestamp with time zone, '1 day')::date
+), curr_archives AS (
+  SELECT start_date FROM archives_archive WHERE org_id = $3 and period = $4 and archive_type=$5
+UNION DISTINCT
+  -- also get the overlapping days for the monthly rolled up archives
```

Comment:
this is a way how we can exclude rolledup monthly archives from missingDailiy archives